velox/type/Type.h

```diff
@@ -435,6 +435,37 @@ struct TypeFactory;
     return this->kind() == TypeKind::KIND;                                \
   }
 
+class Type;
+
+enum class TypeParameterKind {
+  /// Type. For example, element type in the array type.
+  kType,
+  /// Integer. For example, precision in a decimal type.
+  kLongLiteral,
+};
+
+struct TypeParameter {
```

Comment:
how would this work for other parametric types, say a tensor or a fixed list? We would reuse the long literal member?

Reply:
I imagine a fixed list type would have 2 parameters, type of elements and size.